pink/barriers/barrier.py

```diff
@@ -216,8 +234,17 @@ def compute_qp_inequality(
             Tuple containing the inequality constraint matrix (G)
                 and vector (h).
         """
-        G = -self.compute_jacobian(configuration) / dt
-        barrier_value = self.compute_barrier(configuration)
+        if (
+            self.__q_cache is None
+            or self.__jac_cache is None
+            or self.__barrier_cache is None
+        ) or not np.allclose(self.__q_cache, configuration.q):
+            self.__q_cache = configuration.q.copy()
+            self.__jac_cache = self.compute_jacobian(configuration).copy()
+            self.__barrier_cache = self.compute_barrier(configuration).copy()
+
+        G = -self.__jac_cache.copy() / dt
+        barrier_value = self.__barrier_cache.copy()
```

Comment:
Tbh, I have a nasty habit of copying everything and everywhere to avoid those rare but very tricky and disturbing errors with views.

I've double-checked this piece of code, and I believe that it's not necessary at all in both `compute_qp_objective` and `compute_qp_inequality`